src/Finbuckle.MultiTenant/Stores/InMemoryStore/InMemoryStore.cs

```diff
@@ -59,23 +59,24 @@ public async Task<IEnumerable<TTenantInfo>> GetAllAsync()
 
         public async Task<bool> TryAddAsync(TTenantInfo tenantInfo)
         {
-            var result = tenantMap.TryAdd(tenantInfo.Identifier, tenantInfo);
+            // Throws an ArgumentNullException if the identifier is null.
```

Comment:
Would prefer not to throw here if possible -- "Try" functions generally don't throw.

Reply:
I agree, nice catch. I updated it to return false if the identifier is null rather than throw. I also added a test for that scenario.